core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -42,7 +42,7 @@ import scala.util.Try
 
 object KafkaController extends Logging {
   val InitialControllerEpoch = 1
-  val InitialControllerEpochZkVersion = 1
+  val InitialControllerEpochZkVersion = 0
```

Comment:
I think the `InitialControllerEpochZkVersion` represents the initial value when the controller epoch znode first gets created so it should be set to 0 for consistency just like `InitialControllerEpoch`. 

Also, during cluster initialization (i.e controller epoch znode does not exists), we explicitly set `ControllerContext.epochZkVersion` to `InitialControllerEpochZkVersion`, which was 1 before.
(https://github.com/apache/kafka/pull/5101/files/dcf94b0e78208ec1fc11ebe538f743063880a798#diff-ed90e8ecc5439a5ede5e362255d11be1L634)
This will cause problems for the first controller to update zk states after this patch because the actual znode version is 0. This is not a problem before because we didn't use `ControllerContext.epochZkVersion` to fence zk state updates and after controller failover, we will update it by reading controller epoch znode.

This change is only for readbility. Actually we can just remove the line to set `ControllerContext.epochZkVersion` to `InitialControllerEpochZkVersion` and keep `InitialControllerEpochZkVersion` to be 1.

Reply:
Cool. This makes sense.